packages/kbn-migrations/README.md

```diff
@@ -0,0 +1,74 @@
+# Migrations
+
+A concern was raised with the previous approach to migrations: they would allow us to make breaking changes to our data between minor version changes. This might break customers who are consuming data from our Kibana index.
```

Comment:
We could allow breaking changes, and leave it up to the code-review / triage process to determine when / whether to do so. (e.g. If something is experimental, that team can make breaking changes between minor versions, but non-experimental features can't).

I prefer this, as it prevents the migration system from getting in the way when it arguably shouldn't. We would need to underscore w/ tech leads that they should prefer data stability as much as possible, etc.

Reply:
This just doesn't seem like its worth the effort to build and maintain when its so rarely going to be acceptable that we break these APIs on minors. I originally wrote "will never be acceptable", but that's perhaps a bit too absolutist.

Compatibility layers can take us _very_ far with essentially the same impact on consuming code as migrations without any of the risk or complexity of actually moving around all the bits in storage. I'd argue that except for the implications for searching functionality in something like `savedObjects:find`, maintaining migrations and maintaining a compatibility layer are essentially the same thing.

Even experimental features should have graceful paths forward where possible.